scripts/cmake/MCWAMP.cmake

```diff
@@ -55,7 +58,7 @@ macro(add_mcwamp_library_hsa name )
   CMAKE_FORCE_CXX_COMPILER("${PROJECT_BINARY_DIR}/compiler/bin/clang++" MCWAMPCC)
   # add HSA headers
   add_library( ${name} SHARED ${ARGN} )
-  target_include_directories(${name} PRIVATE ${HSA_HEADER})
+  target_include_directories(${name} SYSTEM PUBLIC ${HSA_HEADER})
```

Comment:
Because `hsa_atomic.h` is used be `hc.hpp`: https://github.com/RadeonOpenCompute/hcc/blob/clang_tot_upgrade/include/hc.hpp#L24

So the include needs to be public, so consumers of the target gets the correct includes as well.